src/common/catalog/catalog-entity.ts

```diff
@@ -230,7 +231,21 @@ export abstract class CatalogEntity<
     return this.metadata.name;
   }
 
+  /**
+	 * Trigger when user click on the icon in details panel
+	 * 
+	 * @remarks
+	 * Priority < this.onClickDetailIcon, if this.onClickDetailIcon presents, onRun won't be called.
+	 */
   public abstract onRun?(context: CatalogEntityActionContext): void | Promise<void>;
   public abstract onContextMenuOpen(context: CatalogEntityContextMenuContext): void | Promise<void>;
+
+	/**
+	 * Trigger when user click on the icon in details panel
+	 * 
+	 * @remarks
+	 * Priority > this.onRun, if presents, onRun won't be called.
+	 */
+	public abstract onClickDetailIcon?(context: CatalogEntityActionContext): void;
```

Comment:
I disagree that this is the best API for this, especially since https://github.com/lensapp/lens/issues/3909 has now been asked and because I assume you want to do the work before any `onRun` call.

Furthermore, "detail icon" is too much of an implementation detail and locks us in one design.